src/Compilers/CSharp/Portable/Symbols/SignatureOnlyMethodSymbol.cs

```diff
@@ -35,13 +36,16 @@ internal sealed class SignatureOnlyMethodSymbol : MethodSymbol
             ImmutableArray<TypeParameterSymbol> typeParameters,
             ImmutableArray<ParameterSymbol> parameters,
             RefKind refKind,
+            bool isInitOnly,
             TypeWithAnnotations returnType,
             ImmutableArray<CustomModifier> refCustomModifiers,
             ImmutableArray<MethodSymbol> explicitInterfaceImplementations)
         {
+            Debug.Assert(isInitOnly == CustomModifierUtils.HasIsExternalInitModifier(returnType.CustomModifiers));
```

Comment:
>Debug.Assert(isInitOnly == CustomModifierUtils.HasIsExternalInitModifier(returnType.CustomModifiers)); [](start = 12, length = 102)

Not quite the assert I suggested to add, but, if it holds, let's keep it this way.